distribution/lib/Standard/Database/0.0.0-dev/src/Errors.enso

```diff
@@ -134,3 +134,32 @@ type Table_Not_Found
     to_display_text self = case self.treated_as_query of
         True -> "The name " + self.name + " was treated as a query, but the query failed with the following error: " + self.related_query_error.to_display_text + "; if you want to force to use that as a table name, wrap it in `SQL_Query.Table_Name`."
         False -> "Table " + self.name + " was not found in the database."
+
+type Table_Already_Exists
+    ## PRIVATE
+       Indicates that a table already exists in the database.
+
+       Arguments:
+       - table_name: The name of the table that already exists.
+    Error table_name:Text
+
+    ## PRIVATE
+       Pretty print the table already exists error.
+    to_display_text : Text
+    to_display_text self = "Table " + self.table_name.pretty + " already exists in the database."
+
+type Non_Unique_Primary_Key
+    ## PRIVATE
+       Indicates that the columns selected for the primary key do not uniquely
+       identify rows in the table.
+
+       Arguments:
+       - primary_key: The primary key that is not unique.
+       # TODO [RW] should we include `duplicate_rows` here to show example ids of rows that are not unique?
+    Error (primary_key : Vector Text)
```

Comment:
cc @jdunkerley I wanted to figure out if we want to have these `duplicate_rows`.
It could make the user's life easier to show them indices of rows of one of the duplicated groups.

OTOH, the user can always compute these themselves and computing them is pretty costly for big tables.

Third thing is, when trying to compute them I realised it's actually not _that_ easy. I'm thinking if we should add some methods that could help. I'm thinking of `Aggregate_Column.As_Vector` which could create a vector of all elements of a single group. Or something else that could help - as currently finding out duplicated rows is not that easy - unless I'm not seeing some cool trick.

Reply:
I think this would be very useful; I would think you could do it with a Vector.group_by on the rows, returning a Vector of Vectors, and show the inner vectors with > 1 row. Assign this to me if you like.

Reply:
Ok, I reported this as #6437 and we can handle it when we have a bit of spare time.

> returning a Vector of Vectors

I guess that may require extending `Aggregate_Column`, but yeah that sounds like a good idea.